service/service.go

```diff
@@ -35,6 +35,7 @@ const (
 var (
 	serviceStateLocker stateLocker
 	uuidMapLocker      sync.Map
+	generalRWLocker    sync.RWMutex
```

Comment:
In my opinion, the variable name is very _generalized_.

Reply:
what do you suggest? should i make it specific to secret? it was intended for use with services that doesn't require locking in the UUID

Reply:
yes